makefiles/utils.inc.mk

```diff
@@ -0,0 +1,87 @@
+# Assorted make utilities
+# TODO: move tests out of here
+
+test-all-utils: test-streq test-assert-positive test-assert_eq-positive
+
+# Return an empty string if the arguments are different (works only for single
+# words
+streq = $(filter $(1),$(2))
+
+test-streq:
+	@echo empty,empty: $(call streq,,)
+	@echo empty,x: $(call streq,,x)
+	@echo x, empty: $(call streq,x,)
+	@echo x, y: $(call streq,abc,abcw)
+	@echo y, x: $(call streq,abcw,abc)
+	@echo x,x: $(call streq,abcw,abcw)
+
+# Produce an error if the argument is empty. The second argument defines the
+# message.
+assert = $(if $(1),,$(error $(2)))
```

Comment:
This is now the similar to `ensure_value` from https://github.com/RIOT-OS/RIOT/pull/10342 so definitely putting ithem in common would help :)

Reply:
I think assert can be made an alias for "ensure_value"